openedx/core/djangoapps/programs/utils.py

```diff
@@ -383,27 +384,33 @@ def _attach_run_mode_course_url(self, run_mode):
         run_mode['course_url'] = reverse('course_root', args=[self.course_key])
 
     def _attach_run_mode_end_date(self, run_mode):
-        run_mode['end_date'] = self.course_overview.end_datetime_text()
+        run_mode['end_date'] = self.course_overview.end
 
     def _attach_run_mode_enrollment_open_date(self, run_mode):
         run_mode['enrollment_open_date'] = strftime_localized(self.enrollment_start, 'SHORT_DATE')
 
     def _attach_run_mode_is_course_ended(self, run_mode):
-        end_date = self.course_overview.end or datetime.datetime.max.replace(tzinfo=pytz.UTC)
-        run_mode['is_course_ended'] = end_date < timezone.now()
+        end_date = self.course_overview.end or datetime.datetime.max.replace(tzinfo=utc)
+        run_mode['is_course_ended'] = end_date < datetime.datetime.now(utc)
 
     def _attach_run_mode_is_enrolled(self, run_mode):
         run_mode['is_enrolled'] = CourseEnrollment.is_enrolled(self.user, self.course_key)
 
+    def _attach_run_mode_user_preferences(self, run_mode):
+        run_mode['user_preferences'] = get_user_preferences(self.user)
```

Comment:
You're probably right. The user preferences DO need to be sent to the js functions, and this was an easy way to do it. If you've got another suggestion, I'm happy to implement it.